Wikipedia/Code/StorageAndSyncingSettingsViewController.swift

```diff
@@ -0,0 +1,213 @@
+        let eraseSavedArticles = StorageAndSyncingSettingsItemType.eraseSavedArticles.settingsItem()
+        let syncWithTheServer = StorageAndSyncingSettingsItemType.syncWithTheServer.settingsItem()
+        
+        let syncSavedArticlesAndListsSection = SettingsSection(headerTitle: nil, footerText: WMFLocalizedString("settings-storage-and-syncing-enable-sync-footer-text", value: "Allow Wikimedia to save your saved articles and reading lists to your user preferences when you login to sync", comment: "Footer text of the settings option that enables saved articles and reading lists syncing"), items: [syncSavedArticlesAndLists])
+        
+        let showSavedReadingListSection = SettingsSection(headerTitle: nil, footerText: WMFLocalizedString("settings-storage-and-syncing-show-default-reading-list-footer-text", value: "Show the Saved (eg. default) reading list as a separate list in your Reading lists view. This list appears on Android devices", comment: "Footer text of the settings option that enables showing the default reading list"), items: [showSavedReadingList])
+        
+        let eraseSavedArticlesSection = SettingsSection(headerTitle: nil, footerText: nil, items: [eraseSavedArticles])
+        let syncWithTheServerSection = SettingsSection(headerTitle: nil, footerText: WMFLocalizedString("settings-storage-and-syncing-server-sync-footer-text", value: "Request a sync from the server for an update to your synced articles and reading lists", comment: "Footer text of the settings button that initiates saved articles and reading lists server sync"), items: [syncWithTheServer])
+        
+        return [syncSavedArticlesAndListsSection, showSavedReadingListSection, eraseSavedArticlesSection, syncWithTheServerSection]
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = WMFLocalizedString("settings-storage-and-syncing-title", value: "Saved articles storage and syncing", comment: "Title of the saved articles storage and syncing settings screen")
+        tableView = UITableView(frame: view.bounds, style: .grouped)
+        tableView.contentInset = UIEdgeInsets(top: 10, left: 0, bottom: 0, right: 0);
+        tableView.separatorStyle = .none
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.register(WMFSettingsTableViewCell.wmf_classNib(), forCellReuseIdentifier: WMFSettingsTableViewCell.identifier())
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: customViewCellReuseIdentifier)
+        view.wmf_addSubviewWithConstraintsToEdges(tableView)
+        apply(theme: self.theme)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadRows(at: indexPathsForCellsWithSwitches, with: .none)
+    }
+    
+    private var isSyncEnabled: Bool {
+        guard let dataStore = dataStore else {
+            return false
```

Comment:
Consider an assertionFailure msg here if we don't expect this to ever happen but would like to know if it starts happening.